packages/std/src/sections.rs

```diff
@@ -6,6 +8,33 @@ pub fn decode_sections2(data: Vec<u8>) -> (Vec<u8>, Vec<u8>) {
     (first, second)
 }
 
+/// Encodes multiple sections of data into one vector.
+///
+/// Each section is suffixed by a section length encoded as big endian uint32.
+/// Using suffixes instead of prefixes allows reading sections in reverse order,
+/// such that the first element does not need to be re-allocated if the contract's
+/// data structure supports truncation (such as a Rust vector).
+///
+/// The resulting data looks like this:
+///
+/// ```ignore
+/// section1 || section1_len || section2 || section2_len || section3 || section3_len || …
+/// ```
+#[allow(dead_code)] // used in Wasm and tests only
+pub fn encode_sections(sections: &[Vec<u8>]) -> Vec<u8> {
```

Comment:
I don't think we need the owned vector here. We can do 

```suggestion
pub fn encode_sections(sections: &[&[u8]]) -> Vec<u8> {
```

and below: 

```rust
out_data.extend_from_slice(section);
```

Reply:
Yes, you're right. This is already addressed in relation to the deserialization changes, above.